### PR TITLE
meta iterators refactor

### PR DESCRIPTION
**Pull request by bekzod**
refactored `_getInherited ` and `for each iterators` to accept list of keys

### CODE REVIEW

File: packages/ember-metal/lib/meta.js [hunk 1]
```diff
@@ -183,27 +183,55 @@ export class Meta {
     return this[key] || (this[key] = Object.create(null));
   }
 
-  _getInherited(key) {
+  _forEachInherited() {
```
Comment (bekzod) [inline packages/ember-metal/lib/meta.js]: accepts list of keys, when `obj.prop1.prop2.prop3` `_forEachInherited('obj','prop1','prop2','prop3', fn)` will call fn with value of `prop3` with current meta and its parent

File: packages/ember-metal/lib/meta.js [hunk 2]
```diff
@@ -183,27 +183,55 @@ export class Meta {
...
+          }
+        }
       }
+
       pointer = pointer.parent;
     }
   }
 
-  _findInherited(key, subkey) {
+  _findInherited() {
```
Comment (bekzod) [inline packages/ember-metal/lib/meta.js]: accepts list of keys, when `obj.prop1.prop2.prop3` `_findInherited('obj','prop1','prop2','prop3')` will find first `prop3` property that has `not undefined` value and returns it

**Comment by krisselden** [comment]
what is the motivation?

**Comment by krisselden** [comment]
Also, this has to be benchmarked using a macro bench, this code though for sure can use some cleanup.  I'm surprised seen is using Object.create(null).

**Comment by bekzod** [comment]
motivation was just consolidate repeating logic, `Object.create(null)` was used before so kept it, will benchmark it

**Comment by bekzod** [comment]
benchmark: https://cl.ly/3T3X3a0V2T2O/results.pdf

**Comment by bekzod** [comment]
closing since it performed bad in benchmarks

**Pull request closed without merging**